syntax = "proto2";

// Reports the encoder ticks for each arm joint
message ArmEncodersMessage {
    required int32 shoulderLiftTicks    = 1;
    required int32 shouldSwivelTicks    = 2;
    required int32 elbowLiftTicks       = 3;
    required int32 elbowSwivelTicks     = 4;
    required int32 wristLiftTicks       = 5;
    required int32 wristSwivelTicks     = 6;
}

message DriveEncodersMessage {  
    optional int32 leftSpeed            = 1;        
    optional int32 rightSpeed           = 2;
    required int32 timestamp            = 3;
}

message RequestMessage {  
    required bool  requestSpeed         = 1;
    required bool  requestDiagnostics   = 2;
    optional int32 leftSpeed            = 3;        
    optional int32 rightSpeed           = 4;
    optional int32 duration             = 5;
    required int32 timestamp            = 6;
}

message StatusLightCommand {
    optional int32 color = 1;
    optional int32 display = 2;
}

message NewStatusLightCommand {
    optional int32 redEnabled = 1;
    optional int32 blueEnabled = 2;
    optional int32 greenEnabled = 3;
    optional int32 redBlink = 4;
    optional int32 blueBlnk = 5;
    optional int32 grenBlink = 6;
}

message ArmClawRequest {
    optional int32 clawVel = 1;
}

message ArmEffortRequest {
    optional int32 shoulderLiftEffort  = 1;
    optional int32 shoulderSwivelEffort = 2;
    optional int32 elbowLiftEffort      = 3;
    optional int32 wristLiftEffort      = 4;
    optional int32 wristSwivelEffort    = 5;
    optional int32 clawVel              = 6;
}

message ArmPositionFeedback {
    optional int32 shoulderLiftTicks    = 1;
    optional int32 shoulderSwivelTicks  = 2;
    optional int32 elbowLiftTicks       = 3;
    optional int32 wristLiftTicks       = 4;
    optional int32 wristSwivelTicks     = 5;
}

message ScienceMotorRequest {
    optional int32 leadscrewVel         = 1;
    optional int32 turntableVel         = 2;
    optional int32 drillEffort          = 3;
}
